src/components/todo/TodoItem.tsx

```diff
@@ -16,25 +20,84 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
     setTodos(todos);
   };
 
+  // State to check edit state(if already in edit mode)
+  const [editMode, setEditMode] = useState<boolean>(false);
+  //state to store edit value
+  const [editTodo, setEditTodo] = useState<string>(todoItem.todo);
```

Comment:
I guess a todo is an object which contains all the data related to that item

Reply:
No, todo is just a field name of type string inside TodoModal interface